nix/yazi-unwrapped.nix

```diff
@@ -1,67 +1,76 @@
 {
-  makeRustPlatform,
-  rustToolchain,
+  rustPlatform,
   version ? "git",
-  rev,
-  date,
+  rev ? "unknown",
+  date ? "19700101",
   lib,
 
   installShellFiles,
   stdenv,
   darwin,
+  rust-jemalloc-sys,
 
   imagemagick,
 }:
+let
+  src = lib.fileset.toSource {
+    root = ../.;
+    fileset = lib.fileset.intersection (lib.fileset.fromSource (lib.sources.cleanSource ../.)) (
+      lib.fileset.unions [
+        ../assets
+        ../Cargo.toml
+        ../Cargo.lock
+        (lib.fileset.fileFilter (file: file.hasExt "rs" || file.hasExt "toml") ../.)
```

Comment:
There are also `.lua` files should be included. I think we should add all `yazi-*` directories instead of filter by extension name.

Reply:
when testing i found no `.lua` files. But if we did want to do `yazi-*` it would look something like this.

```nix
src = lib.fileset.toSource {
    root = ../.;
    fileset = lib.fileset.unions (
      builtins.concatLists [
        [
          ../assets
          ../Cargo.toml
          ../Cargo.lock
        ]

        (map (name: ../. + "/${name}") (
          builtins.filter (name: lib.hasPrefix "yazi-" name) (builtins.attrNames (builtins.readDir ../.))
        ))
      ]
    );
  };
```

Which is not awfully pleasant

Reply:
There are many `.lua` files in `yazi-plugin/preset`.

What about this?

```nix
  src = lib.fileset.toSource {
    root = ../.;
    fileset = lib.fileset.intersection (lib.fileset.fromSource (lib.sources.cleanSource ../.)) (
      lib.fileset.unions [
        ../assets
        ../Cargo.toml
        ../Cargo.lock
        (lib.fileset.fromSource (lib.sources.sourceByRegex ../. [ "^yazi-.*" ]))
      ]
    );
  };
```